doc/tutorials/acrn-dm_QoS.rst

```diff
@@ -0,0 +1,119 @@
+and dependencies together. Multiple containers can run on the same machine
+and share the OS kernel with other containers, each running as
+isolated processes in user space. runC, a lightweight universal container runtime,
+is a command-line tool for spawning and running containers according
+to the Open Container Initiative (OCI) specification.
+
+ACRN-DM QoS architecture
+************************
+In ACRN-DM QoS design, we run the ACRN-DM in a runC container environment.
+Every time we start a UOS, we run a runC container at first;
+then launch the ACRN-DM in the started container. The resources
+the ACRN-DM QoS can manage for Device-Model:
+
+- CPU utilization
+- Memory amount/limitation
+- I/O bandwidth
+- Network throughput
+
+.. figure:: images/acrn-dm_qos_architecture.png
+   :align: center
+
+   ACRN-DM QoS architecture
+
+ACRN-QoS CPU utilization example
+********************************
+In runC ``config.json`` we set the CPU resource as below for VM0 and VM1:
+
+.. code-block:: none
+
+    "cpu": {
+       "shares": 1024,
+       "quota": 1000000,
+       "period": 500000,
+       "realtimeRuntime": 950000,
+       "realtimePeriod": 1000000,
+       "cpus": "2-3",
+       "mems": "0-7"
+    },
+
+In this example the CPU shares value is 1024, so the VM0 or VM1 device model
+CPU utilization is ``1024 / (1024 + 1024 +1024) = 33%``, which means
+the maximal CPU resource for the VM0 or VM1 is 33% of the entire CPU resource.
+
+.. figure:: images/cpu_utilization_image.png
+   :align: center
+
+   CPU utilization image
+
+How to use ACRN-DM QoS
+**********************
+
+ #. Add these parameters to the ``runC.json`` file:
+
+    .. code-block:: none
+
+        # vim /usr/share/acrn/samples/nuc/runC.json
+
+    .. code-block:: none
+
+        "linux": {
+            "resources": {
+                "memory": {
+                     "limit": 536870912,
+                     "reservation": 536870912,
+                     "swap": 536870912,
+                     "kernel": -1,
+                     "kernelTCP": -1,
+                     "swappiness": 0,
+                     "disableOOMKiller": false
+                },
+                "cpu": {
+                    "shares": 1024,
+                    "quota": 1000000,
+                    "period": 500000,
+                    "realtimeRuntime": 950000,
+                    "realtimePeriod": 1000000,
+                    "cpus": "2-3",
+                    "mems": "0-7"
+                },
+                "devices": [
+                    {
+                        "allow": true,
+                        "access": "rwm"
+                    }
+                ]
+            },
+
+    .. note:: For the configuration details you can refer to the `config.md
+        <https://github.com/opencontainers/runtime-spec/blob/master/config.md>`_
+
+ #. Add the UOS by ``acrnctl add`` command:
+
+    .. code-block:: none
+
+        # acrnctl add launch_uos.sh -C
```

Comment:
Can you describe what the `-C` argument means? I did not find it in our online documentation.

Reply:
We add the "-C" parameter for the ACRN-DM QoS function.  when we add the luanch script with "-C" parameter means we will start this VM in the container .

Reply:
Can you add this to our `acrnctl` documentation?

Reply:
Hi Geoffroy,
 I will add this argument to acrnctl  documentation.

Thanks